nixos/modules/services/network-filesystems/kubo.nix

```
@@ -211,6 +232,28 @@ in
   ###### implementation
 
   config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !builtins.hasAttr "Identity" cfg.settings;
+        message = ''
+          You can't set services.kubo.settings.Identity.
+          Setting services.kubo.settings.Identity.PeerID would
+          break the Kubo configuration and setting
+          services.kubo.settings.Identity.PrivKey would prevent
+          the daemon from starting with
+          "Error: setting private key with API is not supported".
+        '';
+      }
+      {
+        assertion = !((builtins.hasAttr "Pinning" cfg.settings) && (builtins.hasAttr "RemoteServices" cfg.settings.Pinning));
+        message = ''
+          You can't set services.kubo.settings.Pinning.RemoteServices
+          as it would result in an error message when the daemon starts:
+          "Error: cannot add or remove remote pinning services with 'config replace'".
```

```suggestion
          You can't set services.kubo.settings.Pinning.RemoteServices because the ``config replace`` subcommand used at startup does not work with it.
```

The erorr message in itself is not the problem, lets make that clear.

---

Good suggestion.